src/Aspire.Hosting.Azure.Sql/AzureSqlExtensions.cs

```diff
@@ -148,7 +150,7 @@ public static IResourceBuilder<AzureSqlDatabaseResource> AddDatabase(this IResou
     /// builder.Build().Run();
     /// </code>
     /// </example>
-    public static IResourceBuilder<AzureSqlServerResource> RunAsContainer(this IResourceBuilder<AzureSqlServerResource> builder, Action<IResourceBuilder<SqlServerServerResource>>? configureContainer = null)
+    public static IResourceBuilder<AzureSqlServerResource> RunAsContainer(this IResourceBuilder<AzureSqlServerResource> builder, Action<IResourceBuilder<SqlServerServerResource>>? configureContainer = null, Action<RunAsContainerOptions>? configureOptions = null)
```

Comment:
Adding a default parameter is a breaking change, which is why APICompat is failing. The reason why it is a breaking change is that if someone compiled against the older version (without the default parameter) would be bound to a method that no longer exist, so you'd get a MissingMethodException when running against this change.

Reply:
Just add a new overload. The existing one can then invoke it with a `null` callback.